libgalois/include/katana/LC_CSR_CSC_Labeled_Graph.h

```diff
@@ -337,34 +337,47 @@ class LC_CSR_CSC_Labeled_Graph
       GraphNode value =
           in_edges ? this->getInEdgeDst(mid) : this->getEdgeDst(mid);
       if (value == key) {
-        // Update bounds for last only if we're searching for the first element.
         if (finding_first) {
+          // Update bounds for last only if we're searching for the first
+          // element. We know mid has value == key, so the last element has to
+          // be at or after mid. The begin bound is inclusive so we inlude mid
+          // here.
           last_begin_bound = mid;
-          last_end_bound = r + 1;
         }
 
-        if (!(opt == FindEdgeWithLabelOptions::FindAny) && mid != limit) {
+        // If we're just searching for any valid edge, then we don't need to
```

Comment:
can you file an issue to make a test for this?

Reply:
to make unit tests*

Reply:
Just to be clear. This entire class will be superseded by PropertGraph soon. We should unit tests for this function then.
@amberhassaan